pagopa/pn-frontend-e2e
# PN-9109-refactor-dei-cookie-su-tutti-i-test-e-2-e

### Issue Description
A configuration for cookies has never been added. As a result, every time a test (a new session) was initiated, it had to automatically click 'accept' again. This process generates a spam of logs on the OneTrust side.

Unfortunately, after careful analysis, I noticed that Selenium, through the webdriver tool, does not allow injecting saved cookies for different domains, only for the domain it is currently testing.

Here are a couple of posts complaining about the design that webdriver adopted:

- https://github.com/w3c/webdriver/issues/1238#issuecomment-848251901
- https://stackoverflow.com/questions/22507204/selenium-cookie-with-another-domain


With this modification, I have created a list of specific URLs. For each request, a check is performed using the key (URL). If it corresponds to the domain, the necessary cookies are injected.

### Changes Made:

- added a `CookieConfig `class to manage cookie settings and inject them when necessary
- added on `Hooks` class a check on each request in order to inject cookies
- modified specific test steps in certain methods to check if the environment property cookie.enabled is set to true


### How to Test
Run any test and check that cookies are injected and the OneTrust banner is not shown by adding `-Dcookie.config=true` to the command line.
